plugin/hosts/hostsfile.go

```diff
@@ -68,11 +68,17 @@ type Hostsfile struct {
 func (h *Hostsfile) ReadHosts() {
 	now := time.Now()
 
+	// Not deferring h.RUnlock() because we may need to remove the read lock and aquire a write lock
+	h.RLock()
 	if now.Before(h.expire) && len(h.byAddr) > 0 {
+		h.RUnlock()
 		return
 	}
 	stat, err := os.Stat(h.path)
 	if err == nil && h.mtime.Equal(stat.ModTime()) && h.size == stat.Size() {
+		h.RUnlock()
+		h.Lock()
+		defer h.Unlock()
```

Comment:
This `defer` is a bit superfluous as your return 2 lines below, I would just sandwitch the `h.expire ...` line between lock and unlock

Reply:
Sure thing